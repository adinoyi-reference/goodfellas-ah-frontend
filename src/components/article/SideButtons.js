import React, { Component } from 'react';
import { connect } from 'react-redux';
import icons from '../../assets/icons.svg';

class SideButtons extends Component {
  state = {
    extraButtons: false,
  }

  showExtraButtons = () => {
    this.setState((prevState) => ({ extraButtons: !prevState.extraButtons }));
  }

  AddExtraButtons = (className) => (
    <div className={`extra-buttons ${className}`}>
      <svg id="delete" className="icon" title="Delete this article"><use xlinkHref={`${icons}#delete-icon`} /></svg>
      <svg id="edit" title="Edit this article"><use xlinkHref={`${icons}#edit`} /></svg>
      <svg id="flag" title="Report this article"><use xlinkHref={`${icons}#flag`} /></svg>
      <svg id="dislike" title="Dislike this article"><use xlinkHref={`${icons}#dislike`} /></svg>
    </div>
  )


  render() {
    const { extraButtons } = this.state;
    console.log(this.props.article.slug);
    return (
      <div className="side-button">
        <svg id="star" title="Like this article"><use xlinkHref={`${icons}#star`} /></svg>
        <svg id="bookmark" onClick={() => console.log('I clicked here')} title="Report this article"><use xlinkHref={`${icons}#bookmark`} /></svg>
        <svg id="share" title="Report this article"><use xlinkHref={`${icons}#share`} /></svg>
        <svg id="more" onClick={this.showExtraButtons} title="Show more buttons"><use xlinkHref={`${icons}#more`} /></svg>
        {this.AddExtraButtons(extraButtons ? 'active' : '')}
        <div id="like-count"><p>4</p></div>
        <svg id="love" title="Report this article"><use xlinkHref={`${icons}#gratipay`} /></svg>
      </div>
    );
  }
}

// const mapStateToProps = (state) => ({
//   error: state.articles.articleError,
//   bookmarked: state.articles.article,
// });

export default connect(null)(SideButtons);
