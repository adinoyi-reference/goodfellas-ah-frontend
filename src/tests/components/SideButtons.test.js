import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SideButtons from '../../components/article/SideButtons';

let wrapped;

beforeEach(() => {
  wrapped = shallow(<SideButtons />);
});


describe('SideButtons UI', () => {
  it('view should render as expected', () => {
    const tree = toJson(wrapped);
    expect(tree).toMatchSnapshot();
  });
});
