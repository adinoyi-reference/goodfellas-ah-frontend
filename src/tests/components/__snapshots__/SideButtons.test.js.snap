// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideButtons UI view should render as expected 1`] = `
<div
  className="side-button"
>
  <svg
    id="star"
    title="Like this article"
  >
    <use
      xlinkHref="icons.svg#star"
    />
  </svg>
  <svg
    id="bookmark"
    title="Report this article"
  >
    <use
      xlinkHref="icons.svg#bookmark"
    />
  </svg>
  <svg
    id="share"
    title="Report this article"
  >
    <use
      xlinkHref="icons.svg#share"
    />
  </svg>
  <svg
    id="more"
    onClick={[Function]}
    title="Show more buttons"
  >
    <use
      xlinkHref="icons.svg#more"
    />
  </svg>
  <div
    className="extra-buttons "
  >
    <svg
      className="icon"
      id="delete"
      title="Delete this article"
    >
      <use
        xlinkHref="icons.svg#delete-icon"
      />
    </svg>
    <svg
      id="edit"
      title="Edit this article"
    >
      <use
        xlinkHref="icons.svg#edit"
      />
    </svg>
    <svg
      id="flag"
      title="Report this article"
    >
      <use
        xlinkHref="icons.svg#flag"
      />
    </svg>
    <svg
      id="dislike"
      title="Dislike this article"
    >
      <use
        xlinkHref="icons.svg#dislike"
      />
    </svg>
  </div>
  <div
    id="like-count"
  >
    <p>
      4
    </p>
  </div>
  <svg
    id="love"
    title="Report this article"
  >
    <use
      xlinkHref="icons.svg#gratipay"
    />
  </svg>
</div>
`;
